mk/rt.mk

```diff
@@ -223,147 +223,366 @@ endif
+      i386/ashrdi3.o \
+      i386/chkstk.o \
+      i386/chkstk2.o \
+      i386/divdi3.o \
+      i386/floatdidf.o \
+      i386/floatdisf.o \
+      i386/floatdixf.o \
+      i386/floatundidf.o \
+      i386/floatundisf.o \
+      i386/floatundixf.o \
+      i386/lshrdi3.o \
+      i386/moddi3.o \
+      i386/muldi3.o \
+      i386/udivdi3.o \
+      i386/umoddi3.o
 endif
 
-ifndef COMPRT_DIR_$(1)
-# NB: FreeBSD and NetBSD output to "linux"...
-COMPRT_DIR_$(1) := linux
-COMPRT_ARCH_$(1) := $$(patsubst i586,i386,$$(COMPRT_ARCH_$(1)))
+else
 
-ifeq ($$(findstring android,$(1)),android)
-ifeq ($$(findstring arm,$$(COMPRT_ARCH_$(1))),arm)
-COMPRT_ARCH_$(1) := armhf
-endif
+ifeq ($$(findstring x86_64,$(1)),x86_64)
+COMPRT_OBJS_$(1) += \
+      x86_64/floatdidf.o \
+      x86_64/floatdisf.o \
+      x86_64/floatdixf.o
 endif
 
-ifeq ($$(findstring eabihf,$(1)),eabihf)
-ifeq ($$(findstring armv7,$(1)),)
-COMPRT_LIB_NAME_$(1) := clang_rt.builtins-armhf
-endif
 endif
 
-ifndef COMPRT_LIB_NAME_$(1)
-COMPRT_LIB_NAME_$(1) := clang_rt.builtins-$$(COMPRT_ARCH_$(1))
-endif
+# Generic ARM sources
+ifeq ($$(findstring arm,$(1)),arm)
+COMPRT_OBJS_$(1) += \
+  arm/aeabi_cdcmp.o \
+  arm/aeabi_cdcmpeq_check_nan.o \
+  arm/aeabi_cfcmp.o \
+  arm/aeabi_cfcmpeq_check_nan.o \
+  arm/aeabi_dcmp.o \
+  arm/aeabi_div0.o \
+  arm/aeabi_drsub.o \
+  arm/aeabi_fcmp.o \
+  arm/aeabi_frsub.o \
+  arm/aeabi_idivmod.o \
+  arm/aeabi_ldivmod.o \
+  arm/aeabi_memcmp.o \
+  arm/aeabi_memcpy.o \
+  arm/aeabi_memmove.o \
+  arm/aeabi_memset.o \
+  arm/aeabi_uidivmod.o \
+  arm/aeabi_uldivmod.o \
+  arm/bswapdi2.o \
+  arm/bswapsi2.o \
+  arm/clzdi2.o \
+  arm/clzsi2.o \
+  arm/comparesf2.o \
+  arm/divmodsi4.o \
+  arm/divsi3.o \
+  arm/modsi3.o \
+  arm/switch16.o \
+  arm/switch32.o \
+  arm/switch8.o \
+  arm/switchu8.o \
+  arm/sync_synchronize.o \
+  arm/udivmodsi4.o \
+  arm/udivsi3.o \
+  arm/umodsi3.o
 endif
 
-
-ifeq ($$(findstring windows-gnu,$(1)),windows-gnu)
-COMPRT_LIB_FILE_$(1) := lib$$(COMPRT_LIB_NAME_$(1)).a
+# Thumb sources
+ifeq ($$(findstring armv7,$(1)),armv7)
+COMPRT_OBJS_$(1) += \
+  arm/sync_fetch_and_add_4.o \
+  arm/sync_fetch_and_add_8.o \
+  arm/sync_fetch_and_and_4.o \
+  arm/sync_fetch_and_and_8.o \
+  arm/sync_fetch_and_max_4.o \
+  arm/sync_fetch_and_max_8.o \
+  arm/sync_fetch_and_min_4.o \
+  arm/sync_fetch_and_min_8.o \
+  arm/sync_fetch_and_nand_4.o \
+  arm/sync_fetch_and_nand_8.o \
+  arm/sync_fetch_and_or_4.o \
+  arm/sync_fetch_and_or_8.o \
+  arm/sync_fetch_and_sub_4.o \
+  arm/sync_fetch_and_sub_8.o \
+  arm/sync_fetch_and_umax_4.o \
+  arm/sync_fetch_and_umax_8.o \
+  arm/sync_fetch_and_umin_4.o \
+  arm/sync_fetch_and_umin_8.o \
+  arm/sync_fetch_and_xor_4.o \
+  arm/sync_fetch_and_xor_8.o
 endif
 
-ifeq ($$(findstring android,$(1)),android)
-ifeq ($$(findstring arm,$(1)),arm)
-COMPRT_LIB_FILE_$(1) := $$(call CFG_STATIC_LIB_NAME_$(1),$$(COMPRT_LIB_NAME_$(1))-android)
+# VFP sources
+ifeq ($$(findstring eabihf,$(1)),eabihf)
```

Comment:
@parched sure! I'm hoping to land this in *any* form first (just mirroring what we have today), but this should be an easy follow-up PR